### PR TITLE
Adding useQuery search URL params hook

### LINKED ISSUE #41: Add 'anchor' for the batch viewer into the page URL
originally reported in https://github.com/gnosis/gp-ui/pull/1061#issuecomment-1076358189

It would be nice tp add some kind of an anchor into the URL to separate what Tx view we are using.
I mean, if we see Batch viewer, add #batch into the URL or so?
So a user will be able to share this link and open exactly the viewer, not the list. Also, when a user updates the page, it will also show the viewer.
https://watch.screencastify.com/v/V0tvqMqCZzgtUIRcsfnK

Updated:
- We will improve the link to the Batch Viwer by adding a toogle button or a tab (see screenshots on comments)

> Original issue https://github.com/gnosis/gp-ui/issues/1072 by @elena-zh

### PR DESCRIPTION
**Pull request by henrypalacios**
# Summary
Closes #41 

Adding a hook that parses the search parameter query `tab` in the URL to differentiate **which tab view to display**, `orders` or `graph` nodes.
 

# To Test

1. Go to [/tx/HashId?tab=graph](https://pr49--gpui.review.gnosisdev.com/tx/0xc2eb04b549d68c770062591e2561c5b1534ea665c6179d33a779c52376c9161d?tab=graph)`
and you will see the graph instead of the order list.

### CODE REVIEW

**Comment by ramirotw** [comment]
I don't know what do you think @henrypalacios @alongoni, but I like more how the graph looks like when it fills the container space without any borders

image

image

**Review by elena-zh — APPROVED** [review]
Awesome!

**Review by ramirotw — APPROVED** [review]

File: src/apps/explorer/components/TransanctionBatchGraph/elementsBuilder.tsx [hunk 1]
```diff
@@ -159,9 +163,10 @@ export function buildGridLayout(
   const _nodes = nodes.map((node) => {
     const _node = {
       ...node,
-      position: {
-        y: counterRows[node.data.type],
-        x: getGridPosition(node.data.type, traders, dexes),
+      data: {
+        ...node.data,
+        row: counterRows[node.data.type],
+        col: getGridPosition(node.data.type, traders, dexes),
```
Comment (ramirotw) [inline src/apps/explorer/components/TransanctionBatchGraph/elementsBuilder.tsx:169]: this was causing the layout issue? nice finding!!!

Comment (henrypalacios) [reply src/apps/explorer/components/TransanctionBatchGraph/elementsBuilder.tsx:169]: Yes, I was using the position `propesties` but this layout uses that attribute to position on the **x** and **y** axis when [defining the column/row](https://github.com/cowprotocol/explorer/blob/41-anchor-url/src/apps/explorer/components/TransanctionBatchGraph/index.tsx#L181).

File: src/apps/explorer/components/TransactionsTableWidget/index.tsx [hunk 2]
```diff
@@ -24,30 +25,52 @@ interface Props {
...
 
 export const TransactionsTableWidget: React.FC<Props> = ({ txHash }) => {
   const { orders, isLoading: isTxLoading, errorTxPresentInNetworkId, error } = useGetTxOrders(txHash)
   const networkId = useNetworkId() || undefined
   const [redirectTo, setRedirectTo] = useState(false)
-  const [batchViewer, setBatchViewer] = useState(false)
+  const { tab } = useQueryViewParams()
+  const [tabViewName, setTabViewName] = useState<TabViews>(
+    (Object.values(TabViews).includes(tab as TabViews) && (tab as TabViews)) || TabViews.ORDERS,
+  )
```
Comment (alfetopito) [inline src/apps/explorer/components/TransactionsTableWidget/index.tsx:67]: It's an enum, you can access it via `TabViews[tab]`

```suggestion
  const [tabViewName, setTabViewName] = useState<TabViews>(TabViews[tab] || TabViews.ORDERS)
```

File: src/apps/explorer/components/TransactionsTableWidget/index.tsx [hunk 3]
```diff
@@ -24,30 +25,52 @@ interface Props {
   transactions?: Order[]
 }
 
-const tabItems = (isLoadingOrders: boolean): TabItemInterface[] => {
+enum TabViews {
```
Comment (alfetopito) [inline src/apps/explorer/components/TransactionsTableWidget/index.tsx:28]: I think this should be in the singular

File: src/apps/explorer/components/TransactionsTableWidget/index.tsx [hunk 4]
```diff
@@ -60,6 +83,18 @@ export const TransactionsTableWidget: React.FC<Props> = ({ txHash }) => {
     return (): void => clearTimeout(timer)
   })
 
+  const onChangeTab = useCallback((tabId: number) => {
+    const newTabViewName = Object.keys(TAB_VIEW_ID).find((key) => TAB_VIEW_ID[key as TabViews] === tabId)
```
Comment (alfetopito) [inline src/apps/explorer/components/TransactionsTableWidget/index.tsx:87]: I feel like the TAB_VIEW_ID is not really needed.

What about you instead create the enum without string keys.

See https://stackoverflow.com/a/55578603/1272513 for reference of what I mean

**Review by alfetopito — APPROVED** [review]
Very nice!

Left some refactoring suggestions

File: src/apps/explorer/components/TransactionsTableWidget/index.tsx (hunk 4 above)
Comment (henrypalacios) [reply src/apps/explorer/components/TransactionsTableWidget/index.tsx]: Here if I didn't get it 🤔, 
if I create the *enum* without string like:

```typescript
enum TabView {
  GRAPH,
  ORDERS,
}
```

then I would need a way to get the `tabNumberId` from the **string Name** (`tab` [URL param](https://github.com/cowprotocol/explorer/blob/41-anchor-url/src/apps/explorer/components/TransactionsTableWidget/index.tsx#L42)) and `tabViewName` from the **identifier number** ([onChangeTab callback](https://github.com/cowprotocol/explorer/blob/41-anchor-url/src/apps/explorer/components/TransactionsTableWidget/index.tsx#L84)) to update the URL params.

Comment (alfetopito) [reply src/apps/explorer/components/TransactionsTableWidget/index.tsx]: I've created this [ts playground](https://www.typescriptlang.org/play?#code/FAehAIEkDNwTwPYFdwFskGcAu5sEMAnHLACwFNwBLAOwBMyAPcaAhVcARiuuzL1vAJYABgA04AFaYcqPAGsKpCowAOAG0oBjLJXB4MGSgHNqqMtSzBzSdgBU8AIwBqlMgHdwAb2Dhf4AOIASgCCAAoAEuAAvJyiPn4A8oEAIgCigQDK0eAATMAAvsCgEPoYNjRG4KSUGFS1biR4OIgoRmQ4LGxV5OBIBGrAmgg8xI4Acnhm2QDkRgR4KiTTRWDgALJ4cA6K5ARk07V44G6bVQjgtOekTceUpMjEPZr6OtRG4g5IOHjUcCdw4mqtUuZHqjRwbgoJwsg2GGAQajIADo1AgjAAKezOVxuADaWHGk2RWAQAFUVCoyAQAML6MjogCUelqSGocmoCDc1GZ4CxLncAF0GSsIIisAcqgQ4HpwM8MBQOR4GlSdhQFNKQRhqNMcIwajgaN1lNQbLCRsdWG8sRMpjFphyCQSHMshjwEcjURi+TjcW5LUZrUSeaz2ZzufpeY5+W4hSLmWVUBVujU6sdwamhom3sxWOwlFVHFQBM81GoHHhNHIzdgCw5IAIYhwiq74YiUWjMVGfU767HisnaimGjc3Mg1AJtrgSXsBJRYJgk-nrOxaJQ9to1HBq6M67QAGK51Im9gxb3uJFBMLhZtw93tr1d9z4xz1g9sI82PtAA) explaining my thinking here

You can run to see the results and play around with it

One thing to keep in mind is that TS enums are mapped to 2 way data bindings on JS code.
So you can access the values both by key and value to get the opposite:

```ts
enum E {
  value = 0
}

E['value'] === 0 // === E.value
E[0] === 'value'
```

Comment (henrypalacios) [reply src/apps/explorer/components/TransactionsTableWidget/index.tsx]: Got it.
Thanks for the details!

**Pull request merged**